### DISCUSSION TITLE
support multiple configuration files

### DISCUSSION THREAD

**Discussion by umputun** [TwiN/gatus#297] [Ideas]
For a large system having a lot of endpoints can quickly become large and hard to maintain. If we can split those into parts, i.e. "commons.yml" for the general configuration and a bunch of "systemN.yml" configs, this could help greatly to navigate and maintain endpoints.

Technically it will need to accept the configuration directory in addition to the configuration file. To simplify things it can combine all the files prior to parsing/validation internally, so from gatus point of view it will stay a single "file"

The part about live reload probably needs some change as well to trigger not on a single file but on any file in the configuration directory

**Comment by TwiN** [comment]
To be honest, if the size of the configuration becomes so large that you need to split it in different files, you'll run into other issues anyways, so you might as well have a different dashboard for each file 😅

I do want to add support for paging, after which a change like this would make more sense, but at the moment, merging conflicts between multiple config files is something I'd like to avoid, as it makes it unnecessarily more complex for developers to navigate through the configuration.

**Comment by TwiN** [comment]
This has been implemented some time ago, see [ `linux/arm64` (armv8) and](https://github.com/TwiN/gatus/blob/master/README.md#usage).

As of this message, this is the relevant documentation:

> By default, the configuration file is expected to be at config/config.yaml.
> 
> You can specify a custom path by setting the `GATUS_CONFIG_PATH` environment variable.
> 
> If `GATUS_CONFIG_PATH` points to a directory, all `*.yaml` and `*.yml` files inside said directory and its subdirectories are merged like so:
> 
> - All maps/objects are deep merged (i.e. you could define `alerting.slack` in one file and `alerting.pagerduty` in another file)
> - All slices/arrays are appended (i.e. you can define `endpoints` in multiple files and each endpoint will be added to the final list of endpoints)
> - Parameters with a primitive value (e.g. `debug`, `metrics`, `alerting.slack.webhook-url`, etc.) may only be defined once to forcefully avoid any ambiguity
>   - To clarify, this also means that you could not define `alerting.slack.webhook-url` in two files with different values. All files are merged into one before they are processed. This is by design.
